tethne/serialize/paper.py

```diff
@@ -0,0 +1,178 @@
+                paper.volume = ""
+            if not hasattr(paper, 'title'):
+                paper.title = ""
+            if not hasattr(paper, 'date'):
+                paper.date = ""
+            if self.source == 3:
+                self.paperIdMap[paper.wosid] = pid
+            elif self.source == 2:
+                self.paperIdMap[paper.url] = pid
+
+        if self.corpus is not None:
+                papersDetails = [{
+                    "model": "django-tethne.paper",
+                    "pk": self.paperIdMap[paper.wosid],
+                    "fields": {
+                    "paper_id": paper.wosid,
+                    "corpus": self.corpus_id,
+                    "pub_date": paper.date,
+                    "volume":paper.volume,
+                    "title" : paper.title,
+                    "abstract":paper.abstract,
+                }
+            } for paper in self.corpus]
+        return papersDetails
+
+    def serializeAuthors(self):
+        author_details = []
+
+        if self.corpus is not None:
+            auid = tethnedao.getMaxAuthorID()
+            for val in self.corpus.features['authors'].index.values():
+                auid = auid+1
+                self.authorIdMap[val[1]+val[0]] = auid
+                instanceData = {
+                    "model": "django-tethne.author",
+                    "pk": auid,
+                    "fields":{
+                        "first_name": val[1],
+                        "last_name": val[0],
+                    }
+                }
+                author_details.append(instanceData)
+        return author_details
+
+
+    def serializeAuthorInstances(self):
+        author_instance_details = []
+        author_identity = []
+        auInstanceId = tethnedao.getMaxAuthorInstanceID()
+
+        if self.corpus is not None:
+            for paper in self.corpus:
+                for author in paper.authors:
+                    auInstanceId = auInstanceId+1
+                    self.authorInstanceIdMap[paper.wosid+author[0][1]+author[0][0]]=auInstanceId
+                    instance_data = {
+                    "model": "django-tethne.author_instance",
+                    "pk": auInstanceId,
+                    "fields":{
+                        "paper": self.paperIdMap[paper.wosid],
+                        "author": self.authorIdMap[author[0][1]+author[0][0]],
+                        "first_name": author[0][1],
+                        "last_name":author[0][0],
+                            }
+                    }
+                    author_instance_details.append(instance_data)
+                    identity_data = {
+                    "model": "django-tethne.author_identity",
+                    "fields" :{
+                        "author": self.authorIdMap[author[0][1]+author[0][0]],
+                        "author_instance": self.authorInstanceIdMap[paper.wosid+author[0][1]+author[0][0]],
+                        "confidence": 0.0,
```

Comment:
Why ``0.0`` ? We are creating an author_identity relation in which we have no confidence? That doesn't make sense.